cl/api/webhooks.py

```diff
@@ -0,0 +1,244 @@
+import json
+
+import requests
+from rest_framework.renderers import JSONRenderer
+from scorched.response import SolrResponse
+
+from cl.alerts.api_serializers import (
+    DocketAlertSerializer,
+    SearchAlertSerializerModel,
+)
+from cl.alerts.models import Alert
+from cl.alerts.utils import OldAlertReport
+from cl.api.models import Webhook, WebhookEvent, WebhookEventType
+from cl.api.utils import (
+    generate_webhook_key_content,
+    update_webhook_event_after_request,
+)
+from cl.celery_init import app
+from cl.corpus_importer.api_serializers import DocketEntrySerializer
+from cl.lib.string_utils import trunc
+from cl.recap.api_serializers import PacerFetchQueueSerializer
+from cl.recap.models import PacerFetchQueue
+from cl.search.api_serializers import SearchResultSerializer
+from cl.search.api_utils import SolrObject
+from cl.search.models import DocketEntry
+
+
+def send_webhook_event(
+    webhook_event: WebhookEvent, content_bytes: bytes | None = None
+) -> None:
+    """Send the webhook POST request.
+
+    :param webhook_event: An WebhookEvent to send.
+    :param content_bytes: Optional, the bytes JSON content to send the first time
+    the webhook is sent.
+    """
+    headers = {
+        "Content-type": "application/json",
+        "Idempotency-Key": str(webhook_event.event_id),
+    }
+    if content_bytes:
+        json_bytes = content_bytes
+    else:
+        renderer = JSONRenderer()
+        json_bytes = renderer.render(
+            webhook_event.content,
+            accepted_media_type="application/json;",
+        )
+    try:
+        response = requests.post(
+            webhook_event.webhook.url,
+            data=json_bytes,
+            timeout=(1, 1),
+            stream=True,
+            headers=headers,
+            allow_redirects=False,
+        )
+        update_webhook_event_after_request(webhook_event, response)
+    except (requests.ConnectionError, requests.Timeout) as exc:
+        error_str = f"{type(exc).__name__}: {exc}"
+        trunc(error_str, 500)
+        update_webhook_event_after_request(webhook_event, error=error_str)
+
+
+@app.task()
+def send_docket_alert_webhooks(
+    des_pks: list[int],
+    webhook_recipients_pks: list[int],
+) -> None:
```

Comment:
I'm curious...we don't usually put tasks outside of celery...I guess our tests should prove that this works, but are you confident it does?

Reply:
Great. Better to be careful. Let's not forget to do the second step here.